tests/MongoDB.Driver.Core.Tests/Core/Compression/CompressorSourceTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using MongoDB.Driver.Core.Compression;
+using MongoDB.Driver.Core.Configuration;
+using Xunit;
+
+namespace MongoDB.Driver.Core.Tests.Core.Compression
+{
+    public class CompressorSourceTests
+    {
+        [Fact]
+        public void Constructor_should_throw_if_the_allowed_compressors_collection_is_null()
+        {
+            var subject = Record.Exception(() => CreateSubject(null));
+            subject
+                .Should().BeOfType<ArgumentNullException>()
+                .Subject.ParamName
+                .Should().Be("allowedCompressors");
+        }
+
+        [Fact]
+        public void Get_should_add_a_new_compressor_to_the_cache_and_return_the_cached_compressors_if_available()
+        {
+            var subject = CreateSubject();
+            var compressor = subject.Get(CompressorType.Zlib);
+            var cachedCompressor = subject.Get(CompressorType.Zlib);
+            compressor.Should().BeSameAs(cachedCompressor);
```

Comment:
Minor comment which I'm sure I've made before. It is helpful in our tests to separate the arrange/act/assert sections by a blank line, as in:

            var subject = CreateSubject();

            var compressor = subject.Get(CompressorType.Zlib);
            var cachedCompressor = subject.Get(CompressorType.Zlib);

            compressor.Should().BeSameAs(cachedCompressor);

Normally the `act` section would be a single line, but sometimes (like here) it can be two lines.